libs/newsletter/src/effects/newsletter.effects.ts

```diff
@@ -0,0 +1,46 @@
+import { Inject } from '@angular/core';
+import { of } from 'rxjs';
+
+import { Actions, Effect, ofType, createEffect } from '@ngrx/effects';
+import { switchMap, map, catchError } from 'rxjs/operators';
+
+import { DaffNewsletterServiceInterface } from '../driver/interfaces/newsletter-service.interface';
+import { DaffNewsletterActionTypes, DaffNewsletterSubscribe, DaffNewsletterSuccessSubscribe, DaffNewsletterFailedSubscribe, DaffNewsletterRetry } from '../actions/newsletter.actions';
+import { DaffNewsletterSubmission } from '../models/newsletter.model';
+import { DaffNewsletterDriver } from '../driver/injection-tokens/newsletter-driver.token';
+
+export class DaffNewsletterEffects<T extends DaffNewsletterSubmission, V>{
+
+  constructor(
+    private actions$: Actions,
+    @Inject(DaffNewsletterDriver) private driver: DaffNewsletterServiceInterface<T, V>) { }
+
+  @Effect()
+  trySubmission$ = createEffect(() => this.actions$.pipe(
```

Comment:
Since these functions do the same thing, would it make sense to use two arguments to `ofType`?

Reply:
I would agree with that, however I wasn't sure if `ofType` could accept two arguments.